packages/dds/tree/src/test/simple-tree/api/schemaFactory.spec.ts

```diff
@@ -543,6 +563,24 @@ describe("schemaFactory", () => {
 			class NamedList extends factory.array("name", factory.number) {}
 			const namedInstance = new NamedList([5]);
 		});
+
+		it("Node schema metadata", () => {
+			const factory = new SchemaFactory("");
+
+			class Foo extends factory.array("Foo", factory.number, {
+				metadata: { description: "An array of numbers", custom: { baz: true } },
```

Comment:
Can metadata be updated after creation? There may be some interesting edge cases where the metadata of a object property can change based on some user scenarios

Reply:
This API already lets you put mutable objects in your metadata, though I'd recommend against such a pattern in most cases since schema having state is likely to cause issues as they are static and stateful static are common sources of issues (like tests modify the state, and it show up for future tests in the run, or two apps share package with stateful statics, and a third app that combines both of them breaks due to them stepping on each other's state.

Reply:
No, it can't be modified after construction. Note that this is metadata for the _schema_ and not for individual objects. View schema can't be changed at runtime (trees can undergo schema upgrades, but that's the closest we get to allowing schema changes). This data isn't persisted, so this isn't a strict requirement, but for consistency I think we need to keep this immutable.

Note that there are other patterns app developers can use to associate mutable metadata with their schemas (storing mutable data on their schema subclasses, separate maps, etc.)